lib/plugins/aws/package/compile/layers/index.js

```diff
@@ -93,14 +101,77 @@ class AwsCompileLayers {
 
     newLayerOutput.Value = { Ref: layerLogicalId };
 
+    const layerHashOutputLogicalId = this.provider.naming.getLambdaLayerHashOutputLogicalId(
+      layerName
+    );
+    const newLayerHashOutput = this.cfOutputLayerHashTemplate();
+    newLayerHashOutput.Value = sha;
+
+    const layerS3KeyOutputLogicalId = this.provider.naming.getLambdaLayerS3KeyOutputLogicalId(
+      layerName
+    );
+    const newLayerS3KeyOutput = this.cfOutputLayerS3KeyTemplate();
+    newLayerS3KeyOutput.Value = newLayer.Properties.Content.S3Key;
+
     _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Outputs, {
       [layerOutputLogicalId]: newLayerOutput,
+      [layerHashOutputLogicalId]: newLayerHashOutput,
+      [layerS3KeyOutputLogicalId]: newLayerS3KeyOutput,
     });
   }
 
+  compareWithLastLayer(layerName) {
+    const stackName = this.provider.naming.getStackName();
+    const layerHashOutputLogicalId = this.provider.naming.getLambdaLayerHashOutputLogicalId(
+      layerName
+    );
+
+    return this.provider
+      .request('CloudFormation', 'describeStacks', { StackName: stackName })
+      .then(data => {
+        const lastHash = _.find(data.Stacks[0].Outputs, {
+          OutputKey: layerHashOutputLogicalId,
+        });
+        const compiledCloudFormationTemplate = this.serverless.service.provider
```

Comment:
style nit: We can exit here if `lastHash` is not found

Reply:
Fixed in https://github.com/serverless/serverless/pull/7680/commits/b5b0ec0ac8722033d6275e334df26f11f84a19e8